math/gfm/math/funcs.d

```diff
@@ -20,15 +20,21 @@ else version( D_InlineAsm_X86_64 )
 }
 
 /// Convert from radians to degrees.
-@nogc T degrees(T)(T x) pure nothrow if (!isIntegral!T)
+@nogc T degrees(T)(in T x) pure nothrow if (!isIntegral!T)
 {
-    return x * (180 / PI);
+    static if (isFloatingPoint!T)
+        return x * T(180 / PI); // Avoid floating point conversion.
+    else
+        return x * (180 / PI);
```

Comment:
This other branch exists in case this function is supposed to work with `__vector`s or custom types that override multiplication.

Reply:
Can you clarify what do you want from GFM? All functions in gfm.math.funcs working with __vector?

Reply:
I'm not sure the functions here are supposed to work with "everything". TBD.

__vector imo could be an implementation detail from gfm.Vmath.vector.Vector someday.

I'd rather have these functions work for float, double, real, and maybe Vector.

Reply:
I think it would make sense to restrict the function to only accept floats (and maybe vectors), but that template constraint didn't already exist so I didn't add it in this PR.

Reply:
Updated the template constraint to only accept floating point numbers or a Vector of floating point numbers.